test/e2e/storage/persistent_volumes.go

```diff
@@ -319,6 +326,218 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
+					},
+				},
+			}
+
+			numPVs, numPVCs := 1, 1
+			pvols, claims, err = e2epv.CreatePVsPVCs(ctx, numPVs, numPVCs, c, f.Timeouts, ns, pvHostPathConfig, pvcConfig)
+			framework.ExpectNoError(err, "Failed to create the requested storage resources")
+
+			ginkgo.By(fmt.Sprintf("Listing all PVs with the labelSelector: %q", volLabel.AsSelector().String()))
+			pvList, err := pvClient.List(ctx, metav1.ListOptions{LabelSelector: volLabel.AsSelector().String()})
+			framework.ExpectNoError(err, "Failed to list PVs with the labelSelector: %q", volLabel.AsSelector().String())
+			gomega.Expect(pvList.Items).To(gomega.HaveLen(1))
+			initialPV := pvList.Items[0]
+
+			ginkgo.By(fmt.Sprintf("Listing PVCs in namespace %q", ns))
+			pvcList, err := pvcClient.List(ctx, metav1.ListOptions{})
+			framework.ExpectNoError(err, "Failed to list PVCs with the labelSelector: %q", volLabel.AsSelector().String())
+			gomega.Expect(pvcList.Items).To(gomega.HaveLen(1))
+			initialPVC := pvcList.Items[0]
+
+			ginkgo.By(fmt.Sprintf("Patching the PV %q", initialPV.Name))
+			payload := "{\"metadata\":{\"labels\":{\"" + initialPV.Name + "\":\"patched\"}}}"
+			patchedPV, err := pvClient.Patch(ctx, initialPV.Name, types.StrategicMergePatchType, []byte(payload), metav1.PatchOptions{})
+			framework.ExpectNoError(err, "Failed to patch PV %q", initialPV.Name)
+			gomega.Expect(patchedPV.Labels[patchedPV.Name]).To(gomega.ContainSubstring("patched"), "Checking that patched label has been applied")
+
+			ginkgo.By(fmt.Sprintf("Patching the PVC %q", initialPVC.Name))
+			payload = "{\"metadata\":{\"labels\":{\"" + initialPVC.Name + "\":\"patched\"}}}"
+			patchedPVC, err := pvcClient.Patch(ctx, initialPVC.Name, types.StrategicMergePatchType, []byte(payload), metav1.PatchOptions{})
+			framework.ExpectNoError(err, "Failed to patch PVC %q", initialPVC.Name)
+			gomega.Expect(patchedPVC.Labels[patchedPVC.Name]).To(gomega.ContainSubstring("patched"), "Checking that patched label has been applied")
+
+			ginkgo.By(fmt.Sprintf("Getting PV %q", patchedPV.Name))
+			retrievedPV, err := pvClient.Get(ctx, patchedPV.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Failed to get PV %q", patchedPV.Name)
+			gomega.Expect(retrievedPV.UID).To(gomega.Equal(patchedPV.UID))
+
+			ginkgo.By(fmt.Sprintf("Getting PVC %q", patchedPVC.Name))
+			retrievedPVC, err := pvcClient.Get(ctx, patchedPVC.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Failed to get PVC %q", patchedPVC.Name)
+			gomega.Expect(retrievedPVC.UID).To(gomega.Equal(patchedPVC.UID))
+
+			ginkgo.By(fmt.Sprintf("Deleting PVC %q", retrievedPVC.Name))
+			err = pvcClient.Delete(ctx, retrievedPVC.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "Failed to delete PVC %q", retrievedPVC.Name)
+
+			ginkgo.By(fmt.Sprintf("Confirm deletion of PVC %q", retrievedPVC.Name))
+			err = wait.PollUntilContextTimeout(ctx, time.Second, f.Timeouts.PVDelete, false, func(ctx context.Context) (bool, error) {
```

Comment:
`WaitFor<something>` isn't a good pattern for new test cases, see https://github.com/kubernetes/kubernetes/issues/115164 for details. please consider using gomega.Consistently to handle this case, https://github.com/kubernetes/kubernetes/pull/114825 is a good example.

Reply:
I'm not sure if my understanding is correct, cc @pohly  for re-checking.

Reply:
This is not quite the same as https://github.com/kubernetes/kubernetes/issues/115164 because it's expecting no error.

However, @carlory is right, polling with gomega.Eventually is better - see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#polling-and-timeouts

Reply:
@carlory @pohly switch the test to use gomega.Eventually